app/scripts/components/facets/facets.services.ts

```diff
@@ -318,7 +318,7 @@ module ngApp.components.facets.services {
       var facetType = 'terms';
       if (_.includes(fieldName, 'datetime')) {
         facetType = 'datetime';
-      } else if (_.some(['_id', '_uuid', 'md5sum'], a => _.includes(fieldName, a))) {
+      } else if (_.some(['_id', '_uuid', 'md5sum', 'file_name'], a => _.includes(fieldName, a))) {
```

Comment:
partially, each field does have a `type`. For `type: long`, facet type will always be range, but for `type: string` sometimes (the ones in the array) an id facet is wanted, sometimes it's a date facet, otherwise it's a term facet.